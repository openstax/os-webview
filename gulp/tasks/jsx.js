const path = require('path');
const gulp = require('gulp');
const config = require('../config');
const pi = require('gulp-load-plugins')({
    pattern: ['gulp-*', 'gulp.*', 'del']
});

function jsx() {
    const configSrcApp = `${config.src}/app`;
    const configDestApp = `${config.dest}/app`;

    return gulp.src(`${configSrcApp}/**/*.jsx`, {
        since: gulp.lastRun(jsx)
    })
    .pipe(pi.sourcemaps.init({loadMaps: true}))
    .pipe(pi.rename((uri) => {
        uri.extname = '.jsx.js';
    }))
    .pipe(pi.babel({
        compact: false,
        presets: ['@babel/preset-react']
    }))
    // prefix the sourcemaps with '../src/' so webpack can find them
    .pipe(pi.sourcemaps.mapSources(function(sourcePath, file) {
      const sourcePathHtml = sourcePath.replace('.jsx.js', '.jsx');
      const rel = path.relative(path.dirname(`${configDestApp}/${sourcePath}`), `${configSrcApp}/${sourcePathHtml}`)
      return rel;
    }))
    .pipe(pi.sourcemaps.write('.'))
    .pipe(gulp.dest(`${configDestApp}`));
}


function watchJsx() {
    gulp.watch([`${config.src}/**/*.jsx`], config.watchOpts, gulp.series(
        jsx
    ));
}

exports.jsx = jsx;
exports.jsx.watch = watchJsx;
