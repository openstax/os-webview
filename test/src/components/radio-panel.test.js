import RadioPanel from '~/components/radio-panel/radio-panel';
import {clickElement} from '../../test-utils';
import RadioPanelJsx from '~/components/radio-panel/radio-panel.jsx';
import {makeMountRender, snapshotify} from '../../helpers/jsx-test-utils.jsx';

const items = [
    {value: '', html: 'View All'},
    {value: 'in-review', html: 'In Review'},
    {value: 'reviewed', html: 'Reviewed'},
    {value: 'corrected', html: 'Corrected'}
];

describe('RadioPanel', () => {
    let changed = false;
    const p = new RadioPanel(items);

    p.on('change', () => {
        changed = true;
    });
    it('creates', () => {
        expect(p).toBeTruthy();
    });
    it('toggles active status on click', () => {
        expect(p.el.classList.contains('active')).toBe(false);
        clickElement(p.el);
        expect(p.el.classList.contains('active')).toBe(true);
    });
});

describe('RadioPanelJsx', () => {
    let selectedValue = null;
    const updateSelected = (value) => {
        selectedValue = value;
    };
    const wrapper = makeMountRender(RadioPanelJsx, {
        items,
        selectedValue,
        updateSelected
    })();
    const buttons = () => wrapper.find('.filter-button');

    it('matches snapshot', () => {
        expect(snapshotify(wrapper)).toMatchSnapshot();
    });
    it('has a button for each item', () => {
        expect(buttons().length).toBe(items.length);
    });
    it('selects by click', () => {
        const someButton = buttons().at(3);

        expect(someButton.text()).toBe('Corrected');
        someButton.simulate('click');
        expect(selectedValue).toBe('corrected');
    });
    it ('indicates selectedValue with aria-pressed attribute', () => {
        wrapper.setProps({selectedValue: items[3].value});
        expect(buttons().at(3).prop('aria-pressed')).toBe(true);
    });
    it('selects by keydown', () => {
        buttons().at(2).simulate('keydown', {key: 'Enter'});
        expect(selectedValue).toBe('reviewed');
    });
});
