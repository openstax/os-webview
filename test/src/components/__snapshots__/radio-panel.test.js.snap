// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioPanelJsx matches snapshot 1`] = `"<div class=\\"filter-button\\" tabindex=\\"0\\" role=\\"button\\" aria-pressed=\\"false\\"><span>View All</span></div><div class=\\"filter-button\\" tabindex=\\"0\\" role=\\"button\\" aria-pressed=\\"false\\"><span>In Review</span></div><div class=\\"filter-button\\" tabindex=\\"0\\" role=\\"button\\" aria-pressed=\\"false\\"><span>Reviewed</span></div><div class=\\"filter-button\\" tabindex=\\"0\\" role=\\"button\\" aria-pressed=\\"false\\"><span>Corrected</span></div>"`;
